### PR TITLE
Covert To Gradle

### PR DESCRIPTION
**Pull request by zromano**
This MR converts our build system from Maven to Gradle.

The advantages of doing this are:
- Gradle has more performant builds
- Gradle is much easier to read and understand
- Gradle tasks are more easily modifiable
- Most importantly: The team likes Gradle 😄 

Things I know are left to do:

- [x] Figure out how to make the docs
- [x] Update the .gitignore for Gradle
- [x] Make jacoco work for code coverage.
  - It's mostly wired in, I just wasn't sure if we include IT tests and SDK tests in the coverage or not? [Note: they are not included in coverage. Pipeline only runs unit tests]
- [x] Make sure the IT tests work (should be able to run `./gradlew clean integrationTest`
  - Note: They work, but `LoggingIT` and `OAuthFlowImplIT` are currently broken due to my previous MR
 - [x] Make sure the SDK tests work (should be able to run `./gradlew clean sdkTest`
- [x] Ensure everything else in the POM is covered. Make sure we support anything the maven plugins were doing
  - [x] Need to sign artifacts
  - [x] Add details from the POM's `developer` tag
  - [x] Check if we need nexus plugin
    - This is for if we want to deploy to a private Nexus repo for SNAPSHOT deploys. We do not currently do this
  - [x] Check what the maven-shade-plugin does
    - Used for building fat jars (https://stackoverflow.com/a/13620420). We never use this currently
  - [x] Check what license-maven-plugin does
    - Updates the license.txt and the headers of files. We don't currently use so we can update later
  - [x] Determine how `.maven.xml` works
    - This isn't used. It used to be used for the pipeline when we used Travis CI. It was just never cleaned up.
- [x] Update our pipeline to use gradle commands
  - [x] Builds
  - [x] Publish Docs
  - [x] Releases
- [x] Remove the POM and other maven related files
- [x] Update docs to refer to Gradle instead of Maven

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+    jUnitJupiterVersion = "5.5.1"
+    jettyServerVersion = "9.4.41.v20210516"
+    commonsIoVersion = "2.7"
+    assertJVersion = "3.24.2"
+}
+
+// Tell Gradle where to pull dependencies from
+// Note: we don't want to add mavenCentral() here so we don't go to the public maven repo
+repositories {
+    mavenCentral()
```
Comment (apederson94) [inline build.gradle:60]: Should this be here given the above comments?

Comment (zromano) [reply build.gradle:60]: the comment shouldn't be there.... copy pasta from an internal repo

Comment (zromano) [reply build.gradle:60]: removed.

**Comment by zromano** [comment]
Looks like I've managed to commit from both of my Github accounts 🤦

**Review by priyankalakhe — APPROVED** [review]

**Pull request merged**